src/python_testing/mdns_discovery/mdns_async_service_info.py

```diff
@@ -0,0 +1,273 @@
+#
+#    Copyright (c) 2024 Project CHIP Authors
+#    All rights reserved.
+#
+#    Licensed under the Apache License, Version 2.0 (the "License");
+#    you may not use this file except in compliance with the License.
+#    You may obtain a copy of the License at
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS,
+#    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    See the License for the specific language governing permissions and
+#    limitations under the License.
+#
+
+
+import asyncio
+import enum
+from ipaddress import IPv4Address, IPv6Address
+from random import randint
+from typing import TYPE_CHECKING, Dict, List, Optional, Set, Union, cast
+
+from zeroconf import (BadTypeInNameException, DNSAddress, DNSOutgoing, DNSPointer, DNSQuestion, DNSQuestionType, DNSRecord,
+                      DNSService, DNSText, RecordUpdateListener, ServiceInfo, Zeroconf, current_time_millis, service_type_name)
+from zeroconf._utils.net import _encode_address
+from zeroconf.const import (_CLASS_IN, _DNS_HOST_TTL, _DNS_OTHER_TTL, _DUPLICATE_QUESTION_INTERVAL, _FLAGS_QR_QUERY, _LISTENER_TIME,
+                            _MDNS_PORT, _TYPE_A, _TYPE_AAAA, _TYPE_SRV, _TYPE_TXT)
+
+int_ = int
+float_ = float
+str_ = str
+
+
+QU_QUESTION = DNSQuestionType.QU
+QM_QUESTION = DNSQuestionType.QM
+_AVOID_SYNC_DELAY_RANDOM_INTERVAL = (20, 120)
+
+
+@enum.unique
+class DNSRecordType(enum.Enum):
+    """An MDNS record type.
+
+    "A" - A MDNS record type
+    "AAAA" - AAAA MDNS record type
+    "SRV" - SRV MDNS record type
+    "TXT" - TXT MDNS record type
+    """
+
+    A = 0
+    AAAA = 1
+    SRV = 2
+    TXT = 3
+
+
+class MdnsAsyncServiceInfo(ServiceInfo):
+    def __init__(
+        self,
+        zc: 'Zeroconf',
+        name: str,
+        type_: str = None,
+        port: Optional[int] = None,
+        weight: int = 0,
+        priority: int = 0,
+        properties: Union[bytes, Dict] = b'',
+        server: Optional[str] = None,
+        host_ttl: int = _DNS_HOST_TTL,
+        other_ttl: int = _DNS_OTHER_TTL,
+        *,
+        addresses: Optional[List[bytes]] = None,
+        parsed_addresses: Optional[List[str]] = None,
+        interface_index: Optional[int] = None,
+    ) -> None:
+        # Accept both none, or one, but not both.
+        if addresses is not None and parsed_addresses is not None:
```

Comment:
where is this condition from?

Reply:
This code is part of a method from a zeroconf class that has been overridden, though this particular code isn't the part being personalized, this executes as is in the zeroconf library overridden (in this case) or not.